BrightID/src/components/EditProfile/socialMediaThunks.ts

```diff
@@ -0,0 +1,106 @@
+import { find, propEq } from 'ramda';
+import { Draft, original } from 'immer';
+import { PayloadAction } from '@reduxjs/toolkit';
+import socialMediaService, { socialMediaUrl } from '@/api/socialMediaService';
+import { BrightIdNetwork, linkAppId } from '@/components/Apps/model';
+import store from '@/store';
+import { selectAllApps } from '@/reducer/appsSlice';
+import {
+  saveSocialMedia,
+  selectSocialMediaById,
+} from '@/reducer/socialMediaSlice';
+import { selectSocialMediaVariationById } from '@/reducer/socialMediaVariationSlice';
+
+export const saveAndLinkSocialMedia =
+  (incomingSocialMedia: SocialMedia) =>
+  async (dispatch: dispatch, getState: getState) => {
+    const prevProfile = selectSocialMediaById(
+      getState(),
+      incomingSocialMedia.id,
+    );
+
+    // First, update locally, so the user doesn't need to wait for
+    // the server to complete the request to see his new profile in the UI.
+    dispatch(saveSocialMedia(incomingSocialMedia));
+
+    const socialMediaVariation = selectSocialMediaVariationById(
+      getState(),
+      incomingSocialMedia.id,
+    );
+    const brightIdSocialAppData: BrightIdSocialAppData = {
+      synced: false,
+      linked: false,
+      contextId: null,
+      token: null,
+      ...prevProfile?.brightIdSocialAppData,
+    };
+    let { synced, token, contextId, linked } = brightIdSocialAppData;
+
+    if (!token) {
+      try {
+        const data = await socialMediaService.createSocialMedia({
+          profile: incomingSocialMedia.profile,
+          variation: socialMediaVariation.id,
+          network: __DEV__ ? BrightIdNetwork.TEST : BrightIdNetwork.NODE,
+        });
+        token = data.token;
+        contextId = data.contextId;
+        synced = true;
+      } catch (e) {
+        console.log(e);
+      }
+    } else if (
+      prevProfile?.profile !== incomingSocialMedia.profile ||
+      !synced
+    ) {
+      synced = false;
+      try {
+        await socialMediaService.updateSocialMedia({
+          token,
+          profile: incomingSocialMedia.profile,
+        });
+        synced = true;
+      } catch (e) {
+        console.log(e);
+      }
+    }
+    if (contextId && !linked) {
+      const appId = socialMediaVariation.brightIdAppName;
+      if (appId) {
+        const apps = selectAllApps(store.getState());
+        const appInfo = find(propEq('id', appId))(apps) as AppInfo;
+        if (appInfo && appInfo.usingBlindSig) {
```

Comment:
Could use `selectBlindSigApps` or better create a new selector like `selectBlindSigAppById`?

Reply:
I copied that from here. Maybe it's better to refactor both places.
https://github.com/BrightID/BrightID/blob/2298ea5d6df78bbff9c84949b24c7e2411f84d3a/BrightID/src/components/Apps/AppsScreen.tsx#L85